### ISSUE TITLE
Incorrect nullability inference in ternary operator with use of local variable

### ISSUE THREAD

**Issue by kindermannhubert** [dotnet/roslyn#54973]
**Version Used**: VS 17.0.0 Preview 2.0

**Steps to Reproduce**:

Methods M1/2/3 are equivalent, but in the last one warning is reported.

```C#
#nullable enable
using System.Diagnostics.CodeAnalysis;
public class C 
{
    static int M1() 
    {
        if (TryGet(out var obj))
        {
            return obj.GetHashCode(); //ok
        }
        else
        {
            return -1;
        }
    }
    
    static int M2() 
    {
        return TryGet(out var obj) ? obj.GetHashCode() : -1; //ok
    }
    
    static int M3() 
    {
        var ok = TryGet(out var obj);
        return ok ? obj.GetHashCode() : -1; //warning CS8602: Dereference of a possibly null reference.
    }
    
    static bool TryGet([NotNullWhen(true)]out object? obj)
    {
        obj = new();
        return true;
    }   
}
```

**Comment by RikkiGibson** [comment]
Thank you for taking the time to file this issue. This behavior is by-design. See #27011.
